app/src/main/java/com/fer1592/k8s_android_console/viewmodel/ClusterViewModel.kt

```diff
@@ -30,44 +32,70 @@ class ClusterViewModel(private val dispatcher: CoroutineDispatcher = Dispatchers
         get() = _requestBearerToken
 
     // Live data to show if connection test was successful
-    private val _connectionTestSuccessful = MutableLiveData<Boolean?>(null)
-    val connectionTestSuccessful: LiveData<Boolean?>
-        get() = _connectionTestSuccessful
+    private val _displayMessage = MutableLiveData<Int?>(null)
+    val displayMessage: LiveData<Int?>
+        get() = _displayMessage
 
     // Live data to show input data errors
     private val _isInputValid = MutableLiveData<Boolean?>(null)
     val isInputValid: LiveData<Boolean?>
         get() = _isInputValid
 
+    // Live data to disable add/update button
+    private val _processingData = MutableLiveData(false)
+    val processingData: LiveData<Boolean>
+        get() = _processingData
+
     // Function that init the viewModel
     fun getCluster(clusterId: Long, authMethods: List<String>, clusterRepository: ClusterRepository = ClusterRepositoryImplementation()) {
         this.clusterId = clusterId
```

Comment:
here for example, it would be great to have this repo injected in the constructor of the class